src/cursor.ts

```diff
@@ -3,21 +3,46 @@ import {
   ReferenceExpression,
   SelectQueryBuilder,
   StringReference,
+  sql,
 } from "kysely";
 
+declare const SIMPLE_COLUMN_DATA_TYPES: readonly [
+  "varchar",
+  "char",
+  "text",
+  "integer",
+  "boolean",
+  "double precision",
+  "decimal",
+  "numeric",
+  "date",
+  "datetime",
+  "time",
+  "timetz",
+  "timestamp",
+  "timestamptz",
+];
+type SimpleColumnDataType = (typeof SIMPLE_COLUMN_DATA_TYPES)[number];
+
+type RequireNullableAndDataType<T> = T &
```

Comment:
Just a utility type to ensure we either send **both** `nullable` and `dataType`, or none of them